tools/wake-format/formatter.h

```diff
@@ -189,9 +189,9 @@ struct IfElseAction {
 
   ALWAYS_INLINE void run(wcl::doc_builder& builder, ctx_t ctx, CSTElement& node) {
     if (predicate(builder, ctx, node)) {
-      builder.append(if_formatter.compose(ctx.sub(builder), node));
+      builder.append(if_formatter.compose(ctx + builder, node));
```

Comment:
Heh...this is clever but I think I'm still in favor of using `sub` for this. The types are different here and this hides an important point